File: app/src/main/java/org/syxc/zhihudaily/api/RxCallback.java
package org.syxc.zhihudaily.api;

import com.squareup.okhttp.Response;

/**
 * RxCallback
 * Created by syxc on 16/1/30.
 */
public interface RxCallback {
  /**
   * Success http response
   *
   * @param response com.squareup.okhttp.Response
   */
  void onSuccess(Response response);

  void onCompleted();

  /**
   * Fail http response
   *
   * @param e java.lang.Throwable
   */
  void onError(Throwable e);
}

File: app/src/main/java/org/syxc/zhihudaily/api/RxCallback.kt
package org.syxc.zhihudaily.api

import com.squareup.okhttp.Response

/**
 * RxCallback
 * Created by syxc on 16/1/30.
 */
interface RxCallback {
  /**
   * Success http response

   * @param response com.squareup.okhttp.Response
   */
  fun onSuccess(response: Response)

  fun onCompleted()

  /**
   * Fail http response

   * @param e java.lang.Throwable
   */
  fun onError(e: Throwable)
}
